### ISSUE TITLE
Add to cart from wish list throws error if not all options are selected PRODUCTION MODE

### ISSUE THREAD

**Issue by TheFrankman** [magento/magento2#10633]
### Preconditions
Magento Enterprise Edition Version : 2.1.7
Environment : Centos6.7 Nginx
PHP Version : 7.0
Theme : Luma
This CAN be replicated on a completely blank instance of M2 with the Luma theme
This error appears in **PRODUCTION MODE ONLY**

### Steps to reproduce
1. Put into production mode
2. Create a bundle product with two required options
3. Add two products to each of these bundle options
4. Navigate to the frontend. Select 1 product from 1 of the options, add that product to your wish list
5. Go to wish list and click add to cart

### Expected result

1. Expected to be taken back to the product to configure your bundle product (as it does in developer mode) 

### Actual result
The following error is thrown : 

```
Fatal error: Uncaught TypeError: Argument 1 passed to Magento\Catalog\Block\Product\View::__construct() must be an instance of Magento\Catalog\Block\Product\Context, 
instance of Magento\Framework\ObjectManager\ObjectManager given, 
called in /var/www/public/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php on line 93 
and defined in /var/www/public/vendor/magento/module-catalog/Block/Product/View.php:80 
Stack trace: 
#0 /var/www/public/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php(93): Magento\Catalog\Block\Product\View->__construct(Object(Magento\Framework\ObjectManager\ObjectManager)) 
#1 /var/www/public/vendor/magento/framework/ObjectManager/Factory/Compiled.php(88): Magento\Framework\ObjectManager\Factory\AbstractFactory->createObject('Magento\\Catalog...', Array) 
#2 /var/www/public/vendor/magento/framework/ObjectManager/ObjectManager.php(57): Magento\Framework\ObjectManager\Factory\Compiled->create('Magento\\Catalog...', Array) 
#3 /var/www/public/vendor/magento/framework/View/Element in /var/www/public/vendor/magento/module-catalog/Block/Product/View.php on line 80
```

This is a video showing this issue in full : https://www.screencast.com/t/YunqToR7

**Comment by orlangur** [comment]
Does not look like production mode due to
> Exceptions are not displayed to the user, exceptions are written to logs only

Are you sure no error happened during http://devdocs.magento.com/guides/v2.1/config-guide/cli/config-cli-subcommands-mode.html#change-to-production-mode ?

**Comment by TheFrankman** [comment]
**My Development Environment**
```
$ bin/magento deploy:mode:show
Current application mode: production. (Note: Environment variables may override this value.)
```

Prior to putting the site in Production mode this functionality worked. I've also recreated the error on one of our live production sites. 

In this case a 500 error is thrown and that stack trace was retrieved from the PHP error logs.

**Comment by orlangur** [comment]
I didn't ask for a `show` command output. I believe something went wrong during `di:compile` step http://devdocs.magento.com/guides/v2.1/config-guide/cli/config-cli-subcommands-compiler.html#config-cli-subcommands-single

**Comment by TheFrankman** [comment]
I have re-ran `setup:di:compile` here is the output

```
$ bin/magento setup:di:compile
Compilation was started.
Interception cache generation... 7/7 [============================] 100% 38 mins 350.0 MiBBMiB
Generated code and dependency injection configuration successfully.
```

As stated, this is a blank Magento installation, I've also recreate the issue on another site that we host.

**Comment by magento-engcom-team** [comment]
@TheFrankman, thank you for your report.
The issue is already fixed in 2.2.0
